scout/server/extensions/bionano_extension.py

```diff
@@ -0,0 +1,203 @@
+"""
+    Connect to BioNano Access server via it's API.
+
+    https://bionano.com/wp-content/uploads/2023/01/30462-Bionano-Access-API-Guide-1.pdf
+"""
+import json
+import logging
+from typing import Dict, List, Optional, Tuple
+
+from scout.utils.convert import call_safe
+from scout.utils.scout_requests import get_request_json, post_data_request_json
+
+LOG = logging.getLogger(__name__)
+
+
+class BioNanoAccessAPI:
+    """Use BioNano Access API to retrieve reports."""
+
+    def __init__(self):
+        self.url = None
+        self.bionano_username = None
+        self.bionano_password = None
+        self.bionano_token = None
+        self.bionano_user_dict = {}
+
+    def init_app(self, app):
+        self.url = app.config.get("BIONANO_ACCESS")
+        self.bionano_username = app.config.get("BIONANO_USERNAME")
+        self.bionano_password = app.config.get("BIONANO_PASSWORD")
+
+        self._get_token()
+
+    def _get_token(self) -> Optional[Dict]:
+        query = f"{self.url}/administration/api/1.4/login?username={self.bionano_username}&password={self.bionano_password}"
+
+        LOG.info("Authenticating for BioNano Access")
+        json_response = get_request_json(query)
+
+        json_content = json_response.get("content")
+        if not json_content:
+            LOG.debug("No token JSON returned. Message: %s", json_response.get("message"))
+            return None
+        LOG.debug("Token request response was %s", json_content)
+        json_output = json_content.get("output")
+
+        if "user" in json_output:
+            self.bionano_user_dict = json_output.get("user")
+        if "token" in json_output:
+            self.bionano_token = json_output.get("token")
+
+        return json_content
+
+    def _get_auth_cookies(self):
```

Comment:
```suggestion
    def _get_auth_cookie(self) -> Dict:
```

Reply:
Why not naming it cookie if its just one?

Reply:
There are five! 😸

Reply:
OMG, yes, then it's cookies!